### ISSUE TITLE
VcfBreakendToReadPair writes SAMRecords extending outside of chromosome boundries

### ISSUE THREAD

**Issue by ayhand** [PapenfussLab/gridss#39]
I'm not sure if this is a bug or mistake of my own. After generating the breakend.vcf I tried using VcfBreakendToReadPair utility:
`java -ea -Xmx16g -cp GRIDSS/target/gridss-0.11.7-jar-with-dependencies.jar au.edu.wehi.idsv.VcfBreakendToReadPair INPUT=sample.gridss.vcf.idsv.working/sample.gridss.vcf.idsv.breakpoint.vcf OUTPUT=output.bam OUTPUT_FILTERED=output_filtered.bam REFERENCE=reference.fasta`
The error I got:

```
[Fri Oct 21 16:29:30 EDT 2016] Executing as XXX@XXX on Linux 2.6.32-573.22.1.el6.x86_64 amd64; Java HotSpot(TM) 64-Bit Server VM 1.8.0_77-b03; Picard version: 0.11.7
INFO    2016-10-21 16:29:40     FixMateInformation      Sorting input into queryname order.
[Fri Oct 21 16:29:41 EDT 2016] au.edu.wehi.idsv.VcfBreakendToReadPair done. Elapsed time: 0.18 minutes.
Runtime.totalMemory()=2595749888
Exception in thread "main" htsjdk.samtools.SAMFormatException: SAM validation error: ERROR: Read name gridss57046, Read CIGAR M operator maps off end of reference
        at htsjdk.samtools.SAMUtils.processValidationErrors(SAMUtils.java:441)
        at htsjdk.samtools.BAMRecord.getCigar(BAMRecord.java:252)
        at htsjdk.samtools.SAMRecord.getAlignmentEnd(SAMRecord.java:595)
        at htsjdk.samtools.SAMRecord.computeIndexingBin(SAMRecord.java:1525)
        at htsjdk.samtools.SAMRecord.isValid(SAMRecord.java:2031)
        at htsjdk.samtools.BAMFileReader$BAMFileIterator.advance(BAMFileReader.java:664)
        at htsjdk.samtools.BAMFileReader$BAMFileIterator.next(BAMFileReader.java:650)
        at htsjdk.samtools.BAMFileReader$BAMFileIterator.next(BAMFileReader.java:620)
        at htsjdk.samtools.SamReader$AssertingIterator.next(SamReader.java:545)
        at htsjdk.samtools.SamReader$AssertingIterator.next(SamReader.java:519)
        at picard.sam.FixMateInformation.doWork(FixMateInformation.java:178)
        at au.edu.wehi.idsv.VcfBreakendToReadPair$FixMate.fix(VcfBreakendToReadPair.java:112)
        at au.edu.wehi.idsv.VcfBreakendToReadPair.writeVisualisationBam(VcfBreakendToReadPair.java:84)
        at au.edu.wehi.idsv.VcfBreakendToReadPair.doWork(VcfBreakendToReadPair.java:51)
        at picard.cmdline.CommandLineProgram.instanceMain(CommandLineProgram.java:208)
        at au.edu.wehi.idsv.VcfBreakendToReadPair.main(VcfBreakendToReadPair.java:116)

```

Additionally, I think there is a mistake in the main page. It says that bwa is the main aligner in somewhere and bowtie2 in somewhere else.
This is a very cool tool. Thank you.

**Comment by d-cameron** [comment]
Are you able to dump the SAM record for the read `gridss57046` ? It looks like an alignment is mapping over the end of a chromosome, but I'm not sure if it's coming from the input data, the realignment, or the assembly.

**Comment by ayhand** [comment]
Hi,
Sorry for late reply. The record gridss57046 is in the VCF. The two break-ends are in different chromosomes. The position of gridss57046o is 162bp before the end of the chromosome. gridss57046h is in somewhere middle.

**Comment by d-cameron** [comment]
Could you run `grep gridss57046o <output.vcf>` and send through the record in question? Of particular interest are the additional VCF columns which help immensely in tracking down what GRIDSS thinks the breakpoint is.

**Comment by ayhand** [comment]
Here are the records in the vcf:
`CM000592.1      5212600 gridss57046o    C       C]CM000595.1:230648]    208.85  .       CIPOS=-161,162;CIRPOS=-162,161;CQ=208.85;EVENT=gridss57046;PARID=gridss57046h;SVTYPE=BND`
`CM000595.1      230648  gridss57046h    A       A]CM000592.1:5212600]   208.85  .       CIPOS=-162,161;CIRPOS=-161,162;CQ=208.85;EVENT=gridss57046;PARID=gridss57046o;SVTYPE=BND`

(Chromosome size of CM000592.1: 5212762)

**Comment by d-cameron** [comment]
Fixed by 2eb233003fc65d20ebf6bdffd2d94b5bd66e6f38
